src/MSLGenerator.cpp

```diff
@@ -222,6 +222,9 @@ void MSLGenerator::Prepass(HLSLTree* tree, Target target, HLSLFunction* entryFun
 
     HLSLType samplerType(HLSLBaseType_Sampler);
 
+    // Reserve buffer 0 for implicit vertex buffer
+    int bufferRegisterOffset = (m_target == Target_VertexShader) ? 1 : 0;
```

Comment:
Ah, glad you added that back. Ideally this should be generator option. Some use cases may have multiple vertex buffers, others may not have any, etc.